### PR TITLE
feat(ui/save-html): add shortName, publishDate to file name

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/w3c/respec/issues/3533
`backlog--`

### CODE REVIEW

**Comment by sidvishnoi** [comment]
Tests failing as https://status.w3.org/incidents/lfwjx05yvldr

File: src/ui/save-html.js [hunk 1]
```diff
@@ -63,38 +63,46 @@ const downloadLinks = [
...
 ];
 
-function toDownloadLink(details) {
+/**
+ * @param {typeof downloadLinks[0]} details
+ */
+function toDownloadLink(details, conf) {
   const { id, href, fileName, title, type } = details;
+  const filename = conf.shortName
+    ? `${conf.shortName}-${ISODate.format(conf.publishDate)}`
```
Comment (marcoscaceres) [inline src/ui/save-html.js:72]: Is publishDate assured? Or should we not added it if missing?

**Review by marcoscaceres — APPROVED** [review]
Worth adding a test?

File: src/ui/save-html.js (hunk 1 above)
Comment (sidvishnoi) [reply src/ui/save-html.js:72]: image
Should exist on each, but maybe good idea to add a failsafe.

**Comment by sidvishnoi** [comment]
> Worth adding a test?

Would need to add tests for rest of the UI stuff too (we've zero UI/pill tests as of now).

File: src/ui/save-html.js [hunk 2]
```diff
@@ -63,38 +63,46 @@ const downloadLinks = [
...
 ];
 
-function toDownloadLink(details) {
+/**
+ * @param {typeof downloadLinks[0]} details
+ */
+function toDownloadLink(details, conf) {
   const { id, href, fileName, title, type } = details;
+  const filename = conf.shortName
+    ? `${conf.shortName}-${ISODate.format(conf.publishDate || new Date())}`
```
Comment (marcoscaceres) [inline src/ui/save-html.js:72]: Maybe format the date as a W3C date? Or does it come out nice? (Sorry not at computer, but can check tomorrow)

Comment (sidvishnoi) [reply src/ui/save-html.js:72]: I looks like: `{shortName}-2021-10-05`, which I think is a good file name.

**Comment by marcoscaceres** [comment]
I might have a little play with this tomorrow. Now I’m kinda wanting to add the specStatus too 🥰

File: src/ui/save-html.js (hunk 2 above)
Comment (marcoscaceres) [reply src/ui/save-html.js:72]: nah, that's good enough 👍

Comment (marcoscaceres) [inline src/ui/save-html.js:72]: I think we want this:

```suggestion
    ? `${conf.specStatus}-${conf.shortName}-${concatDate(conf.publishDate || new Date())}`
```

So then it comes out like this:
Screen Shot 2021-10-06 at 3 37 36 pm

Which was the original motivation.
